### PR TITLE
chore: Documentation updates on the 'fetch-data' example

### PR DESCRIPTION
**Pull request by albertodev01**
_Description of what this PR is changing or adding, and why:_

_Issues fixed by this PR (if any):_

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/cookbook/networking/fetch-data.md [hunk 1]
```diff
@@ -231,8 +231,10 @@ it's not recommended to put an API call in a `build()` method.
 Flutter calls the `build()` method every time it needs
 to change anything in the view,
 and this happens surprisingly often.
-Leaving the `fetch` call in your `build()` method
-floods the API with unnecessary calls and slows down your app.
+Leaving the `fetch` call in your `build()` method would
```
Comment (domesticmouse) [inline src/cookbook/networking/fetch-data.md:234]: Try reading this line out loud. It feels convoluted to me.

Comment (albertodev01) [reply src/cookbook/networking/fetch-data.md]: You're right, I've updated it!

File: src/cookbook/networking/fetch-data.md [hunk 2]
```diff
@@ -231,8 +231,12 @@ it's not recommended to put an API call in a `build()` method.
 Flutter calls the `build()` method every time it needs
 to change anything in the view,
 and this happens surprisingly often.
-Leaving the `fetch` call in your `build()` method
-floods the API with unnecessary calls and slows down your app.
+The `fetchAlbum()` method, if placed inside `build()`, is repeatedly 
+called on each rebuild causing the app to slow down.
+
+Storing the `fetchAlbum()` result in a state variable ensures that
+the `Future` is executed only once and then "cached" for subsequent
```
Comment (domesticmouse) [inline src/cookbook/networking/fetch-data.md:238]: Drop the " from around cached.

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: src/cookbook/networking/fetch-data.md (hunk 2 above)
Comment (albertodev01) [reply src/cookbook/networking/fetch-data.md]: Thank you, done

**Comment by domesticmouse** [comment]
The CI is picking up that the [snippets are out of sync with their source material](https://github.com/flutter/website/runs/4874970926?check_suite_focus=true).

```
=> ERROR: some code excerpts were not refreshed!
The command '/bin/sh -c tool/test.sh --target $TEST_TARGET_CHANNEL --check-links --null-safety' returned a non-zero code: 1
make: *** [Makefile:148: test] Error 1
Error: Process completed with exit code 2.
```

See [refreshing code excerpts](https://github.com/flutter/website#refreshing-code-excerpts) for details on how to update the snippets. I'm guessing you need to find the sample that the code snippets are pulled from and update that sample code so that the snippets line up.

**Comment by albertodev01** [comment]
@domesticmouse I got it right now!

**Review by domesticmouse — APPROVED** [review]

**Review by sfshaza2 — APPROVED** [review]
lgtm! Thanks, @albertodev01 and @domesticmouse!

**Pull request merged**